include/circt/Dialect/FIRRTL/FIRRTLOps.h

```diff
@@ -102,6 +104,12 @@ bool hasDontTouch(Value value);
 /// should prevent certain types of canonicalizations.
 bool hasDontTouch(Operation *op);
 
+/// Scan all the uses of the specified value, checking to see if there is
+/// exactly one connect that sets the value as its destination.  This returns
+/// the operation if found and if all the other users are "reads" from the
+/// value.
+StrictConnectOp getSingleConnectUserOf(Value value);
```

Comment:
For the intended uses, does this care about dominance?  E.g. do users expect that the write is seen by the reads if this returns a writer?

Reply:
On further thinking, there is a difference in this property between whether you are asking about registers or wires.  Regardless, it should be at least mentioned.

Reply:
Yeah good point. I'll add a comment before merging.